core/handler/device/store.go

```diff
@@ -23,11 +24,13 @@ type Store interface {
 	DownlinkQueue(appID, devID string) (DownlinkQueue, error)
 	Set(new *Device, properties ...string) (err error)
 	Delete(appID, devID string) error
+	AttrWhitelist() map[string]bool
 }
 
 const defaultRedisPrefix = "handler"
 const redisDevicePrefix = "device"
 const redisDownlinkQueuePrefix = "downlink"
+const redisUpStreamMetadataKeyWhitelist = ":attr-whitelist"
```

Comment:
1. ok
2. Then 200 ?
3. "builtin" rings good
4. For the sorting, should there be a mechanism that limit the number of key transmitted? Too many values to sort will make the processors spikes. I think gRPC have a maximum size for the packet no ? or brutally trunc if they are too many.